commands/build.go

```diff
@@ -40,8 +41,20 @@ func buildFn(ccmd *cobra.Command, args []string) {
 }
 
 func buildComplete() bool {
+	// check the boxfile to be sure it hasnt changed
 	env, _ := models.FindEnvByID(config.EnvID())
 	box := boxfile.NewFromPath(config.Boxfile())
 
-	return env.UserBoxfile != "" && env.UserBoxfile == box.String()
+	// we need to rebuild if this isnt true without going to check triggers
+	if env.UserBoxfile == "" || env.UserBoxfile != box.String() {
+		return false
+	}
+
+	// now check to see if any of the build triggers have changed
+	for _, trigger := range box.Node("run.config").StringSliceValue("build_triggers") {
```

Comment:
I could be wrong, but it seems like this is going to try to pull the build_triggers from the boxfile in the codebase directly. Since the engines will likely be setting those, we need to be looking in the built boxfile (from the previous build). If there hasn't been a prior build then this won't matter since a build will happen anyway.